t/form_with_fields_passthrough_params.t

```diff
@@ -0,0 +1,246 @@
+#!perl -T
+
+use warnings;
+use strict;
+use Test::More 'no_plan';
+use Test::Fatal;
+use Test::Warnings ':all';
+use Test::Deep;
+use URI::file ();
+
+BEGIN {
+    delete @ENV{qw(PATH IFS CDPATH ENV BASH_ENV)};  # Placates taint-unsafe Cwd.pm in 5.6.1
+    use_ok( 'WWW::Mechanize' );
+}
+
+my $mech = WWW::Mechanize->new( cookie_jar => undef, autocheck => 0, strict_forms => 1, verbose_forms => 1 );
+isa_ok( $mech, 'WWW::Mechanize' );
+my $uri = URI::file->new_abs( 't/form_with_fields.html' )->as_string;
+
+$mech->get( $uri );
+ok( $mech->success, "Fetched $uri" ) or die q{Can't get test page};
+
+{
+    my $test = 'dies with no input';
+    like(
+        exception { my $form = $mech->form_with_fields(); },
+        qr/no fields provided/,
+        $test,
+    );
+}
+
+{
+    my $form;
+    cmp_deeply(
+        [ warnings { $form = $mech->form_with_fields(qw/1b/) } ],
+        [ re(qr/There are 2 forms with the named fields.  The first one was used./) ],
+        'warning on ambiguous match (1)',
+    );
+    isa_ok( $form, 'HTML::Form' );
+    is($form->attr('name'), '1st_form', 'first form matches');
+}
+
+{
+    my $form = $mech->form_with_fields('1b', 'opt[2]');
+    isa_ok( $form, 'HTML::Form' );
+    is($form->attr('name'), '2nd_form', 'second form matches');
+}
+
+{
+    my $form;
+    cmp_deeply(
+        [ warnings { $form = $mech->form_with_fields('4a', '4b') } ],
+        [ re(qr/There are 2 forms with the named fields.  The first one was used./) ],
+        'warning on ambiguous match (2)',
+    );
+    isa_ok( $form, 'HTML::Form' );
+    is($form->attr('name'), '4th_form_1', 'fourth form matches');
+}
+
+{
+    my @forms = $mech->all_forms_with( name => '3rd_form_ambiguous' );
+    is( scalar @forms, 2 );
+    isa_ok( $forms[0], 'HTML::Form' );
+    isa_ok( $forms[1], 'HTML::Form' );
+    is($forms[0]->attr('name'), '3rd_form_ambiguous', 'first result of 3rd_form_ambiguous');
+    is($forms[0]->attr('name'), '3rd_form_ambiguous', 'second result of 3rd_form_ambiguous');
```

Comment:
I'm not sure how this is different from the previous line.

Reply:
Good spot. See above please. This seems to have been broken in form_with_fields.t already, but I missed it. I will fix it in both tests.